src/Bicep.Core/Registry/Oci/IOciArtifactReference.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Bicep.Core.Registry.Oci
+{
+    public interface IOciArtifactReference
+    {
+         /// <summary>
+        /// Gets the registry URI.
+        /// </summary>
+        public string Registry { get; }
+
+        /// <summary>
+        /// Gets the repository name. The repository name is the path to an artifact in the registry without the tag.
+        /// </summary>
+        public string Repository { get; }
+
+        /// <summary>
+        /// Gets the tag. Either tag or digest is set but not both.
+        /// </summary>
+        public string? Tag { get; }
+
+        /// <summary>
+        /// Gets the digest. Either tag or digest is set but not both.
+        /// </summary>
+        public string? Digest { get; }
+
+        /// <summary>
+        /// Gets the artifact ID.
+        /// </summary>
+        public string ArtifactId {get;}
```

Comment:
>ArtifactId [](http://example.com/codeflow?start=22&length=10)

What is an artifact ID? Can you include an example in the comments?

Reply:
I think this is a member that is sourced from `OciArtifactModuleReference.cs` (https://github.com/Azure/bicep/blob/3319ec7e069073fe24dc2408438fca7185bce022/src/Bicep.Core/Modules/OciArtifactModuleReference.cs#L84-L89) it looks like you introduced it in a module to be the following about 2 years ago:
![image](https://github.com/Azure/bicep/assets/9611108/2b251bfd-fb90-4330-b2a8-667b7deba80c)

I am not sure why its necesary vs using `UnqualifiedReference` since they are really the same but left it by inertia. Would you advise I remove it and just use `UnqualifiedReference` ?

Reply:
Heh that's ironic :D. No strong opinion either way.